### PR TITLE
codec/table: check whether range.start<range.end

### PR DESCRIPTION
**Pull request by AndreMouche**
@BusyJay @hicqu PTAL

### CODE REVIEW

File: src/coprocessor/codec/table.rs [hunk 1]
```diff
@@ -611,4 +618,25 @@ mod test {
             assert_eq!(extract_table_prefix(&input).ok().map(From::from), output);
         }
     }
+
+    #[test]
+    fn test_check_table_range() {
+        let small_key = b"t\x80\x00\x00\x00\x00\x00\x00\x01a".to_vec();
+        let big_key = b"t\x80\x00\x00\x00\x00\x00\x00\x01b".to_vec();
```
Comment (BusyJay) [inline src/coprocessor/codec/table.rs]: s/big/large

File: src/coprocessor/dag/executor/index_scan.rs [hunk 2]
```diff
@@ -422,16 +422,15 @@ mod test {
...
+        let r1 = get_idx_range(TABLE_ID, INDEX_ID, 0, (KEY_NUMBER / 2) as i64, unique);
+        let r2 = get_idx_range(
             TABLE_ID,
             INDEX_ID,
             (KEY_NUMBER / 2) as i64,
             i64::MAX,
             unique,
         );
-        wrapper.ranges = vec![r1, r2, r3];
+        wrapper.ranges = vec![r1, r2];
```
Comment (BusyJay) [inline src/coprocessor/dag/executor/index_scan.rs]: Why change this?

File: src/coprocessor/codec/table.rs [hunk 3]
```diff
@@ -70,6 +70,13 @@ pub fn check_table_ranges(ranges: &[KeyRange]) -> Result<()> {
     for range in ranges {
         extract_table_prefix(range.get_start())?;
         extract_table_prefix(range.get_end())?;
+        if range.get_start() > range.get_end() {
```
Comment (BusyJay) [inline src/coprocessor/codec/table.rs]: Is it valid that `range.get_start() == range.get_end()`?

File: src/coprocessor/dag/executor/index_scan.rs (hunk 2 above)
Comment (AndreMouche) [reply src/coprocessor/dag/executor/index_scan.rs]: Since the previous `r1` has been wrongly encoded.

File: src/coprocessor/codec/table.rs (hunk 3 above)
Comment (AndreMouche) [reply src/coprocessor/codec/table.rs]: no

**Comment by BusyJay** [comment]
LGTM

**Comment by AndreMouche** [comment]
@hicqu @hanfei1991 PTAL

**Comment by hicqu** [comment]
LGTM.

**Pull request merged**